In [1]:
import numpy as np
import pandas as pd

In [2]:
data = np.genfromtxt(r"C:\Users\sathw\Downloads\Sample_data.csv",delimiter=',',skip_header=1)
data

array([[ 0., 25., 40.],
       [ 1., 30., 35.],
       [ 2., 28., 38.],
       [ 3., 32., 41.],
       [ 4., 26., 37.],
       [ 5., 29., 36.],
       [ 6., 27., 39.],
       [ 7., 33., 42.],
       [ 8., 31., 38.],
       [ 9., 28., 37.],
       [10., 30., 40.],
       [11., 29., 36.]])

## measures of central tendency:
#### 1) Calculate the mean sales for Product A and Product B. Compare the average monthly sales for the two products and discuss any observed trends.

In [3]:
Product_A = data[:,1]
Product_B = data[:,2]

a = np.mean(Product_A)
b = np.mean(Product_B)

print('the Mean of monthly sales for Product A : ', a) # also the observed trends
print('the Mean of monthly sales for Product B : ', b) # also the observed trends

the Mean of monthly sales for Product A :  29.0
the Mean of monthly sales for Product B :  38.25


#### comparision of mean monthly sales of two products & observations
* the average monthly sales of product A is 29.0 and product B is 38.25
* product A  sales is lower than product B
* The sales of product B is higher 
> product B performs the better sales when compared to product A

#### 2) Find the median sales for each product and discuss how it reflects the central tendency of the sales distribution.

In [4]:
Product_A = data[:,1]
Product_B = data[:,2]

a = np.median(Product_A)
b = np.median(Product_B)

print('the Median sale of each Product A : ', a)
print('the Median sale of each Product B : ', b)

the Median sale of each Product A :  29.0
the Median sale of each Product B :  38.0


#### the way it reflects the central tendancy of the sales distribution
the medians of a and b (29.0 for Product A and 38.0 for Product B) acts as strongly formed indicators of central tendency.so, the median values implies that Product B gain more sales when compared to Product A in the observed data.

#### 3) Identify the mode of the sales data for both Product A and Product B. Discuss the significance of the mode in the context of these products.

In [5]:
import statistics

a = statistics.mode(Product_A)
b = statistics.mode(Product_B)

print('Mode of Sales data for Product A : ', a)
print('Mode of Sales data for Product B : ', b)

Mode of Sales data for Product A :  30.0
Mode of Sales data for Product B :  40.0


#### significance of the mode for product A and product B
* the significance for the Product A is having mode of 30 which denotes 30 is the most occurance of sales figure.
* Similarly, the significance for the Product B is having mode of 40 that denotes 40 is the most common sales value.

#### 4) Calculate the weighted mean of the sales data, assuming each month has an equal weight. Discuss any implications of using weighted mean in this context.

In [6]:
weight=np.ones(12)

weighted_mean_A = weight * Product_A
a = np.mean(weighted_mean_A)
weighted_mean_B = weight * Product_B
b = np.mean(weighted_mean_B)

print('Weighted mean of the sales data of Product A :', a)
print('Weighted mean of the sales data of Product B :', b)

Weighted mean of the sales data of Product A : 29.0
Weighted mean of the sales data of Product B : 38.25


####  implications of using weighted mean
the calculated weighted mean for product A and product B, assuming an equal weight in each month.

## Measures of Dispersion:
#### 1) Compute the range of monthly sales for both Product A and Product B. Discuss how the range provides insights into the variability of sales.

In [7]:
range_product_A = data[:,1].max() - data[:,1].min()
range_product_B = data[:,2].max() - data[:,2].min()

print(range_product_A)
print(range_product_B)

8.0
7.0


#### the range provides insights into the variability of sales.
* the range insights between the max and min values
1) the max:33 min:25 range of product_A = max_value - min_value = 33 - 25 = 08
2) the max:42 min:35 range of product_B = max_value - min_value = 42 - 35 = 07

#### 2) Calculate the variance of the sales data for both products. Discuss what the variance reveals about the spread of sales for each product.

In [11]:
variance_A = np.var(Product_A)
variance_B = np.var(Product_B)

print('variance of the sales for Product A : ', variance_A)
print('variance of the sales for Product A : ', variance_B)

variance of the sales for Product A :  5.166666666666667
variance of the sales for Product A :  4.354166666666667


#### the variance reveals about the spread of sales for each product.¶
* the variance function reveals the spread of sales figure for product A is more from the mean when compared to the Product B
* the variance function of Product B, implies that its sales data points are closer to the mean. it has more consistent and stable sales performance when compared to Product A.

#### 3) Determine the standard deviation of the sales for Product A and Product B. Discuss the significance of standard deviation in understanding the variability in sales.

In [18]:
standard_deviation_A = np.std(Product_A)
standard_deviation_B = np.std(Product_B)

print('standard variance of  sales for Product A : ', standard_deviation_A)
print('standard variance of  sales for Product B : ', standard_deviation_B)

standard variance of  sales for Product A :  2.273030282830976
standard variance of  sales for Product B :  2.0866640042581523


#### the significance of standard deviation in understanding the variability in sales.
* The squareroot of variance is called standard deviation
* The product A is having the higher standard deviation,  which has more consistent and more stable sales with less variability in sales
* The product B is having the lower standard deviation, which has less consistent and less stable sales with more variability in sales

## Relative Measures of Dispersion:
#### 1) Calculate the coefficient of variation for the sales data of Product A. Discuss how this relative measure of dispersion helps in comparing the variability of products with different average sales.

In [32]:
coefficient_variance_product_A = (standard_deviation_A / np.mean(Product_A)) * 100
print('coefficient variation of sales Product A : ', coefficient_variance_product_A)

coefficient variation of sales Product A :  7.838035458037848


#### 2) Compute the relative range for Product B. Discuss how the relative range accounts for the scale of the sales data.

In [30]:
realtive_range_B = (range_product_B / np.mean(Product_B)) * 100
print('Relative Range for sales Product B : ', realtive_range_B)

Relative Range for sales Product B :  18.30065359477124


#### 3) Determine the relative standard deviation for Product A. Interpret the result and discuss its implications for understanding the variability relative to the mean.

In [31]:
realtive_standard_devation_A = (standard_deviation_A / np.mean(Product_A)) * 100
print('realtive standard devation for sales Product A : ', realtive_standard_devation_A)

realtive standard devation for sales Product A :  7.838035458037848
